src/Instrumentation/Distributed.jl

```diff
@@ -53,3 +89,39 @@ Cassette.posthook(::ExtraeCtx, _, ::typeof(process_messages), args...) = emit(Di
 
 Cassette.prehook(::ExtraeCtx, ::typeof(interrupt), args...) = emit(DistributedInterrupt)
 Cassette.posthook(::ExtraeCtx, _, ::typeof(interrupt), args...) = emit(DistributedEnd)
+
+# workload execution
+Cassette.prehook(::ExtraeCtx, ::typeof(Distributed.run_work_thunk), f::Function, args...) = emit(DistributedUsefulWork)
+Cassette.posthook(::ExtraeCtx, _, ::typeof(Distributed.run_work_thunk), f::Function, args...) = emit(DistributedNotUsefulWork)
```

Comment:
If you are not dispatching on an argument, not typing is best practice.

```suggestion
Cassette.prehook(::ExtraeCtx, ::typeof(Distributed.run_work_thunk), f, args...) = emit(DistributedUsefulWork)
Cassette.posthook(::ExtraeCtx, _, ::typeof(Distributed.run_work_thunk), f, args...) = emit(DistributedNotUsefulWork)
```

Reply:
I am dispatching on an argument. The run work thunk function has two methods and right now, for this event I want to only capture the one that has Function as first argument  type, not the other.